.github/workflows/scripts/jira_helper.py

```diff
@@ -0,0 +1,296 @@
+        }
+        payload = json.dumps({"transition": {"id": transition_id}})
+        self._submit_request(method="POST",
+                             endpoint=f'issue/{issue_id}/transitions',
+                             headers=headers,
+                             parameters=None,
+                             data=payload)
+
+    def _close_issue(self):
+        try:
+            issue_id = self._find_jira_issue_by_gh_issue_url()
+            self.logger.debug(f'Closing issue {issue_id}')
+            self._transition_issue(issue_id, self._done_transition_id)
+            self._add_comment_to_issue(
+                issue_id,
+                f"Closing JIRA issue from GitHub issue {os.environ['ISSUE_URL']}"
+            )
+
+        except NoJiraIssueFound:
+            self.logger.debug(
+                'Did not find JIRA issue matching URL, nothing to do')
+
+    def _reopen_issue(self):
+
+        try:
+            issue_id = self._find_jira_issue_by_gh_issue_url()
+            self.logger.debug(f'Found issue {issue_id}')
+            self._transition_issue(issue_id, self._backlog_transition_id)
+            self._add_comment_to_issue(issue_id, "Reopened issue")
+        except NoJiraIssueFound:
+            self.logger.debug(
+                'Did not find JIRA issue matching URL, creating it')
+            issue_id = self._create_issue()
+            issue_comments = json.loads(
+                self._get_gh_output(
+                    f"gh issue view {os.environ['ISSUE_URL']} --json comments")
+            )
+            self.logger.debug(f'Comments: {issue_comments}')
+            for comment in issue_comments['comments']:
+                self._add_comment_to_issue(issue_id, comment['body'])
+
+    def _get_issue_type(self, labels) -> str:
+        if 'kind/bug' in labels:
+            return 'Bug'
+        else:
+            return 'Task'
+
+    def _create_issue(self):
+        try:
+            labels = os.environ['ISSUE_LABELS']
+            self.logger.debug(f'Labels: {labels}')
+        except Exception:
+            pass
+        headers = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+        issue_body = os.environ['ISSUE_BODY']
+        issue_url = os.environ['ISSUE_URL']
+        issue_title = os.environ['ISSUE_TITLE']
+        labels = os.environ['ISSUE_LABELS'].split(',')
+        issue_type = self._get_issue_type(labels)
+        self.logger.debug(f"Creating an issue of type {issue_type}")
+
+        payload = json.dumps({
+            "fields": {
+                "description": f"{issue_body}",
+                "summary": f"{issue_title}",
+                "issuetype": {
+                    "name": f"{issue_type}"
+                },
+                "labels": labels,
+                "project": {
+                    "key": f"{self._project_key}"
+                },
+                "customfield_10052": f"{issue_url}"
```

Comment:
`customfield_10052` I know exactly what this means 😎

Reply:
`External Github Issue Link`

Reply:
Well yes I figured that out, was more complaining about the JIRA API 😄